### ISSUE TITLE
A11y_Playwright.dev_KeyBoard: When we press "ESC" on the "Search field" Key board focus is not remains on "Search" field.

### ISSUE THREAD

**Issue by Raghu-addepalli** [microsoft/playwright.dev#1269]
**"Please do not close this bug. This bug should only be closed by TCS, C+AI Accessibility or the Divisional Driver"**

"[Check out Accessibility Insights](https://nam06.safelinks.protection.outlook.com/?url=https://accessibilityinsights.io/&data=02%7c01%7cv-stfe%40microsoft.com%7cb67b2c4b646d4f9561a208d6f4b5c39b%7c72f988bf86f141af91ab2d7cd011db47%7c1%7c0%7c636965458850501301&sdata=mxhokIKNMb22llsjXHLgU3XZibj1Qfx37rpY4PU2sfE%3D&reserved=0)! - Identify accessibility bugs before check-in and make bug fixing faster and easier."

### GitHub Tags:
#A11yMAS;#A11yTCS;#Playwright releases;#P1_Playwright.dev_Web_Jan2024;#A11ySev2;#AI Limited;#ChromiumEdge;#DesktopWeb;#FTP;#WCAG2.4.3;#Win11;#KeyBoard;#FocusOrder;#P1_WCP;#Closed;

### Environmental Details:
App Name: Playwright.dev
#URL: https://playwright.dev/docs/input

### OS & Browser Details:
Microsoft Windows 11 Enterprise
Version 22621.3007
Microsoft Edge
Version 120.0.2210.144 (Official build) (64-bit)

### Repro Steps:

1. Hit the URL: https://playwright.dev/docs/input
2. TAB to "Search" Field on Top Navigation Bar and Press ENTER.
3. Now Press "ESC" Key.
4. Now check that when we press "ESC" on the "Search field" Key board focus is not remains on "Search" field or not.

### Actual Result:
When we press "ESC" on the "Search field" dialog Key board focus is not remains on "Search" field.

**Note: Similar issue is also observed in below links.**

https://playwright.dev/docs/test-assertions
https://playwright.dev/docs/running-tests
https://playwright.dev/docs/locators
https://playwright.dev/docs/api/class-locator
https://playwright.dev/docs/writing-tests
https://playwright.dev/docs/api/class-playwright
https://playwright.dev/docs/intro
https://playwright.dev/docs/api/class-page

### Expected Result:
When we press "ESC" on the "Search field" dialog Key board focus should remain on "Search" field.

### User Impact:
Key Board Users will face difficulty while accessing the controls if Focus order issue is observed.

### Attachment:

https://github.com/microsoft/playwright.dev/assets/91586743/2d3bc54e-85cc-4190-aee5-b53e8a7ef739

**Comment by mxschmitt** [comment]
This one needs to be fixed in a 3rd-party dependency, I reported it as https://github.com/facebook/docusaurus/issues/9917.

**Comment by Raghu-addepalli** [comment]
@mxschmitt verified the issue in below environment issue is Still reproducing and added attachment below.
https://playwright.dev/docs/test-assertions
https://github.com/microsoft/playwright.dev/assets/91586743/9102e68d-890c-49f0-9ac3-399c7b576341

**Comment by mxschmitt** [comment]
Yes, waiting for a third-party dependency update.

**Comment by mxschmitt** [comment]
This should be fixed, please re-check.

**Comment by Raghu-addepalli** [comment]
verified the issue in below environment issue is Fixed hence closing the bug and added attachment.
https://playwright.dev/docs/test-assertions

https://github.com/microsoft/playwright.dev/assets/91586743/f2b28921-5a4f-41e7-a541-2921e8c48974
